File: references/cellab_devkit-1.0 (1)/ruledefs/owncode.js
    /*

        Dummy rule definition to invoke user-supplied evaluator.
        No lookup table is used.

    */

    rule.worldtype = 13;         // 2D torus world, user evaluator
    rule.ocodereq = "";         // User will override own code request

    function owncode(oldstate) {
        return 0;               // No lookup table
    }

File: references/cellab_devkit-1.0 (1)/rules_java/owncode.java
/*

    Dummy rule definition to invoke user-supplied own code.
    No lookup table is used.

*/

class Owncode extends ruletable {
    void jcruleModes() {
        setWorld(13);           // Own code, torus world
        setOwnCodeRequest("");  // User own code function
    }

    int jcrule(int oldstate) {
        return 0;               // No lookup table
    }
}

public class owncode {
    public static void main(String args[]) {
        (new Owncode()).generateRuleFile("owncode");
    }
}
